### PR TITLE
ARXIVNG-1553: trackback display and redirect

### PR DESCRIPTION
**Pull request by mhl10**
This PR implements three routes:

- `/tb/<arxiv_id>` - display the trackbacks for an individual article. Leverages the `abs` macro to display metadata about the arXiv article.
- `/tb/recent` - display recent trackbacks across all of arXiv
- `/tb/redirect/<trackback_id>/<hashed_document_id>` - link to external trackback URLs

(this branch is currently staged on beta)

Differences with the classic trackback display pages are minor. For comparison:
- https://arxiv.org/tb/0808.4142 
- https://arxiv.org/tb/recent

Screenshots are from NG reimplementation:

screenshot 2019-01-29 14 13 48
screenshot 2019-01-29 14 13 27
screenshot 2019-01-29 15 25 36

### CODE REVIEW

File: browse/templates/tb/macros.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{%- macro generate_trackback_link(trackback, include_posted_date=True) -%}
+<a class="mathjax" rel="external nofollow" href="{{ url_for('browse.tb_redirect', trackback_id=trackback.trackback_id, hashed_document_id=trackback.hashed_document_id) }}">{{ trackback.title|entity_to_utf|truncate(150) }}</a> [{% if trackback.blog_name %}{{ trackback.blog_name|trim|entity_to_utf|truncate(50) }} @ {% endif %}{% if trackback.has_valid_url %}{{ trackback.display_url|truncate(30,False) }}{% else %}INVALID-URL{% endif %}]{% if include_posted_date %} <small>[trackback posted {{ trackback.posted_datetime.strftime('%a, %-d %b %Y %H:%M:%S %Z') }}]</small>{% endif %}
```
Comment (bdc34) [inline browse/templates/tb/macros.html]: This is a very long line. If there's a way to split it up that'd be nice.

File: tests/test_browse.py [hunk 2]
```diff
@@ -33,6 +34,34 @@ def test_home(self):
...
+        rv = self.app.get('/tb/1901.99999')
+        self.assertEqual(rv.status_code, 404)
+
+        rv = self.app.get('/tb/')
+        self.assertEqual(rv.status_code, 404)
+
+        rv = self.app.get('/tb/foo')
+        self.assertEqual(rv.status_code, 404)
+
+    def test_tb(self):
```
Comment (bdc34) [inline tests/test_browse.py]: Is this the same method name as the previous method on line 37? Is that Okay?

File: tests/test_browse.py [hunk 3]
```diff
@@ -33,6 +34,34 @@ def test_home(self):
         self.assertFalse(html.find('h2', text='Test'),
                          "'Test' group should not be shown on homepage")
 
+    def test_tb(self):
+        """Test the /tb/<arxiv_id> page."""
+        rv = self.app.get('/tb/0808.4142')
+        self.assertEqual(rv.status_code, 200)
```
Comment (bdc34) [inline tests/test_browse.py]: It would be good to check for some expected content.

**Comment by bdc34** [comment]
There seems to be an off by one but on the /tb/recent page
screenshot at 2019-02-08 15-33-52

File: tests/test_browse.py (hunk 2 above)
Comment (mhl10) [reply tests/test_browse.py]: Ooops, should have been called `test_tb_recent()`--fixed.

File: browse/templates/tb/macros.html (hunk 1 above)
Comment (mhl10) [reply browse/templates/tb/macros.html]: fixed

**Comment by mhl10** [comment]
> There seems to be an off by one but on the /tb/recent page

I think this may be an artifact of the test data; will check.

**Comment by mhl10** [comment]
> > There seems to be an off by one but on the /tb/recent page
> 
> I think this may be an artifact of the test data; will check.

One of the test `arXiv_trackback_ping` database entries was referencing an invalid document ID. I've corrected and the page is no longer off by one.

**Comment by mhl10** [comment]
Per earlier discussion with @bdc34, I'm going to merge this now that requested changes have been implemented and tests pass.

**Pull request merged**